cluster/cluster.go

```diff
@@ -191,6 +221,46 @@ func (p *Peer) Peers() []*memberlist.Node {
 	return p.mlist.Members()
 }
 
+// Settle waits until the mesh is ready (and sets the appropriate internal state when it is).
+// The idea is that we don't want to start "working" before we get a chance to know most of the alerts and/or silences.
+func (p *Peer) Settle(ctx context.Context, interval time.Duration) {
+	// Inspired from https://github.com/apache/cassandra/blob/7a40abb6a5108688fb1b10c375bb751cbb782ea4/src/java/org/apache/cassandra/gms/Gossiper.java
+	// This is clearly not perfect or strictly correct but should prevent the alertmanager to send notification before it is obviously not ready.
+	// This is especially important for those that do not have persistent storage.
+	const NumOkayRequired = 3
+	level.Info(p.logger).Log("msg", "Waiting for gossip to settle...", "interval", interval)
+	start := time.Now()
+	nPeers := 0
+	nOkay := 0
+	totalPolls := 0
+	for {
+		select {
```

Comment:
Any reason why the select block isn't at the end of the for block? Currently it takes 4 `interval` for the function to return if everything's ok.

Reply:
yes, it is at the beginning because we start with a `Sleep()` in order to have "some" data when we look at `Peers()` for the first time. As you pointed out it means that we always wait once, then we wait enough to have the thing stabilized for three rounds.